src/samplers/interrupt/mod.rs

```diff
@@ -136,25 +138,18 @@ impl Interrupt {
                 );
                 let mut bpf = bcc::BPF::new(&code)?;
 
-                bcc::Kprobe::new()
-                    .handler("hardirq_entry")
-                    .function("handle_irq_event_percpu")
-                    .attach(&mut bpf)?;
-                bcc::Kretprobe::new()
-                    .handler("hardirq_exit")
-                    .function("handle_irq_event_percpu")
-                    .attach(&mut bpf)?;
-                bcc::Tracepoint::new()
-                    .handler("softirq_entry")
-                    .subsystem("irq")
-                    .tracepoint("softirq_entry")
-                    .attach(&mut bpf)?;
-                bcc::Tracepoint::new()
-                    .handler("softirq_exit")
-                    .subsystem("irq")
-                    .tracepoint("softirq_exit")
-                    .attach(&mut bpf)?;
+                // collect the set of probes required from the statistics enabled.
```

Comment:
This looks a lot more like what I'd expect. Perhaps where I asked why it can't be like this for the block io sampler, you can explain why it doesn't look more like this.

Reply:
This is the case for most of the samplers.
However for the io one, I was trying to follow the existing logic. Answered there as well.